Add pre-commit and lint (#346)
mindspore-lab/mindocr: CONTRIBUTING.md

# MindOCR Contributing Guidelines

Contributions are welcome, and they are greatly appreciated! Every little
helps, and credit will always be given.

## Contributor License Agreement

It's required to sign CLA before your first code submission to MindOCR community.

For individual contributor, please refer to [ICLA online document](https://www.mindspore.cn/icla) for the detailed information.

## Types of Contributions

### Report Bugs

Report bugs at https://github.com/mindspore-lab/mindocr/issues.

If you are reporting a bug, please include:

* Your operating system name and version.
* Any details about your local setup that might be helpful in troubleshooting.
* Detailed steps to reproduce the bug.

### Fix Bugs

Look through the GitHub issues for bugs. Anything tagged with "bug" and "help
wanted" is open to whoever wants to implement it.

### Implement Features

Look through the GitHub issues for features. Anything tagged with "enhancement"

> and "help wanted" is open to whoever wants to implement it.

Reviewer: ```suggestion
and "help wanted" is open to whoever wants to fix it.
```

If you are proposing a feature:

* Explain in detail how it would work.
* Keep the scope as narrow as possible, to make it easier to implement.
* Remember that this is a volunteer-driven project, and that contributions are welcome :)

## Getting Started

Ready to contribute? Here's how to set up `mindocr` for local development.

1. Fork the `mindocr` repo on [GitHub](https://github.com/mindspore-lab/mindocr).
2. Clone your fork locally:

   ```shell
   git clone git@github.com:your_name_here/mindocr.git
   ```

   After that, you should add official repository as the upstream repository:

   ```shell
   git remote add upstream git@github.com:mindspore-lab/mindocr
   ```

3. Install your local copy into a conda environment. Assuming you have conda installed, this is how you set up your fork for local development:

   ```shell
   conda create -n mindocr python=3.8
   conda activate mindocr
   cd mindocr
   pip install -e .
   ```

4. Create a branch for local development:

   ```shell
   git checkout -b name-of-your-bugfix-or-feature
   ```

   Now you can make your changes locally.

> 5. When you're done making changes, check that your changes pass the linters and the tests:

Reviewer: ```suggestion
5. When you finish making changes, check that your changes pass the linters and the tests:
```

   To get pre-commit and pytest, just pip install them into your conda environment.

6. Commit your changes and push your branch to GitHub:

   ```shell
   git add .
   git commit -m "Your detailed description of your changes."
   git push origin name-of-your-bugfix-or-feature
   ```

7. Submit a pull request through the GitHub website.

## Pull Request Guidelines

Before you submit a pull request, check that it meets these guidelines:

1. The pull request should include tests.
2. If the pull request adds functionality, the docs should be updated. Put
   your new functionality into a function with a docstring, and add the
   feature to the list in README.md.
3. The pull request should work for Python 3.7, 3.8 and 3.9, and for PyPy. Check
   https://github.com/mindspore-lab/mindocr/actions
   and make sure that the tests pass for all supported Python versions.

## Tips

You can install the git hook scripts instead of linting with `pre-commit run -a` manually.

run flowing command to set up the git hook scripts

```shell
pre-commit install
```

now `pre-commit` will run automatically on `git commit`!

## Releasing

A reminder for the maintainers on how to deploy.

> Make sure all your changes are committed (including an entry in HISTORY.md).

Reviewer: ```suggestion
Make sure all your changes are committed.
```

Reviewer: remove if unnecessary
